src/components/oculus-touch-controls.js

```diff
@@ -0,0 +1,262 @@
+    this.checkIfControllerPresent();
+  },
+
+  onGamepadDisconnected: function (evt) {
+    this.everGotGamepadEvent = true;
+    this.removeControllersUpdateListener();
+    this.checkIfControllerPresent();
+  },
+
+  play: function () {
+    this.checkIfControllerPresent();
+    window.addEventListener('gamepadconnected', this.onGamepadConnected, false);
+    window.addEventListener('gamepaddisconnected', this.onGamepadDisconnected, false);
+    this.addControllersUpdateListener();
+  },
+
+  pause: function () {
+    window.removeEventListener('gamepadconnected', this.onGamepadConnected, false);
+    window.removeEventListener('gamepaddisconnected', this.onGamepadDisconnected, false);
+    this.removeControllersUpdateListener();
+    this.removeEventListeners();
+  },
+
+  updateModel: function () {
+    var objUrl, mtlUrl;
+    if (!this.data.model) { return; }
+    if (this.data.hand === 'right') {
+      objUrl = 'url(' + TOUCH_CONTROLLER_MODEL_OBJ_URL_R + ')';
+      mtlUrl = 'url(' + TOUCH_CONTROLLER_MODEL_OBJ_MTL_R + ')';
+    } else {
+      objUrl = 'url(' + TOUCH_CONTROLLER_MODEL_OBJ_URL_L + ')';
+      mtlUrl = 'url(' + TOUCH_CONTROLLER_MODEL_OBJ_MTL_L + ')';
+    }
+    this.el.setAttribute('obj-model', {obj: objUrl, mtl: mtlUrl});
+  },
+
+  injectTrackedControls: function () {
+    var el = this.el;
+    var data = this.data;
+    var isRightHand = data.hand === 'right';
+
+    // since each hand is named differently, avoid enumeration
+    el.setAttribute('tracked-controls', {
+      id: isRightHand ? 'Oculus Touch (Right)' : 'Oculus Touch (Left)',
+      controller: 0,
+      rotationOffset: data.rotationOffset !== -999 ? data.rotationOffset : isRightHand ? -90 : 90
+    });
+
+    this.updateModel();
+  },
+
+  addControllersUpdateListener: function () {
+    this.el.sceneEl.addEventListener('controllersupdate', this.onControllersUpdate, false);
+  },
+
+  removeControllersUpdateListener: function () {
+    this.el.sceneEl.removeEventListener('controllersupdate', this.onControllersUpdate, false);
+  },
+
+  onControllersUpdate: function () {
+    if (!this.everGotGamepadEvent) {
+      this.checkIfControllerPresent();
+    }
+  },
+
+  onButtonChanged: function (evt) {
+    var button = this.mapping[this.data.hand]['button' + evt.detail.id];
+    var buttonMeshes = this.buttonMeshes;
+    var value;
+    // at the moment, if trigger or grip,
+    // touch events aren't happening (touched is stuck true);
+    // synthesize touch events from very low values
+    if (button === 'trigger' || button === 'grip') {
+      var lastValue = this['last-' + button + '-value'];
+      var lastFakeTouch = false;
+      if (lastValue) { lastFakeTouch = (lastValue >= FAKE_TOUCH_THRESHOLD); }
+      var thisValue = evt.detail.state.value;
+      var thisFakeTouch = false;
+      if (thisValue) { thisFakeTouch = (thisValue >= FAKE_TOUCH_THRESHOLD); }
+      this['last-' + button + '-value'] = thisValue;
```

Comment:
I would store this flags in an object `this.previousButtonValues` instead of directly on `this` and do:
````javascript
this.previousButtonValues[button];
````